### DISCUSSION TITLE
argo-workflows-submit ClusterRole to allow only submitting workflows but not the full argo-workflows-edit permissions

### DISCUSSION THREAD

**Discussion by komapa** [argoproj/argo-helm#2629] [Q&A]
So you have this installed by default https://github.com/argoproj/argo-helm/blob/main/charts/argo-workflows/templates/controller/workflow-aggregate-roles.yaml and we want to allow ServiceAccount to have permissions to submit workflows via `/api/v1/workflows/NAMESPACE/submit`. So my ask here is, do you think you will approve such contribution and is such permission even possible with how argo-workflows work?

Thank you!

**Comment by yu-croco** [comment]
Hi @komapa , thank you for your question.

> do you think you will approve such contribution and is such permission even possible with how argo-workflows work?

We follow [upstream's manifest](https://github.com/argoproj/argo-workflows/blob/v3.5.5/manifests/cluster-install/workflow-controller-rbac/workflow-aggregate-roles.yaml), so if you want to change the definition, please approach to upstream at first.
